### ISSUE TITLE
pt-caney architecture

### ISSUE THREAD

**Issue by cssprad1** [nasa-nccs-hpda/pytorch-caney#33]
Things got a bit messy considering the overall architecture (my fault :) ). Here are some questions/points/rambling that we can continue to work through.

Deps
- Considering move datasets,loaders,training and inference pipelines behind pt-lightning?

Base classes
- Base class for .npy-based datasets (inherits torch dataset?)
- Base class for webdataset datasets (inherits torch dataset?)
- Base class datamodules (inherits pt-lightning datamodule?)
- Lightning module / trainer for each unique pre-training + fine-tuning strategy
      - MAE, MIM, satMAE, etc

Models
- How do we organize different models? (keeping in mind certain pre-training methods need architecture changes)

Views
- How/Do we implement views?

Other
- Are there cases where we want to keep pipelines that use torch training loops? Does pt-lightning allow us to perform more complex training loop actions such as gradient checkpointing, clipping, etc?
- Is this geared for use in training scripts or more for scientists looking for a command-line application?

**Comment by cssprad1** [comment]
See #63 for updated arch. improvements
